packages/suite/src/services/google.ts

```diff
@@ -208,14 +217,28 @@ class Client {
                 });
 
                 const json = await res.json();
-                Client.accessToken = json.access_token;
-                Client.refreshToken = json.refresh_token;
+                if (!json?.access_token || !json?.refreshToken) {
+                    throw new Error('Could not retrieve the tokens.');
+                } else {
+                    Client.accessToken = json.access_token;
+                    Client.refreshToken = json.refresh_token;
+                }
             } catch {
-                Client.authServerAvailable = false;
+                await Client.forceImplicitFlow();
             }
         }
     }
 
+    // when auth server is running, but returns an unexpected response, fall back to implicit flow
+    // TODO: this does not work if browser blocks pop-up windows and it opens two tabs/windows, there could be a better solution
```

Comment:
Is it possible to react on pop-up close? User changes his mind and wants to use another provider instead. But other options are disabled.
![Screenshot 2022-07-14 at 11 13 18](https://user-images.githubusercontent.com/33235762/178947614-6eb51cd2-8373-492a-8cb7-be368376be4b.png)
r

Reply:
I noticed it stops loading with a small delay after to pop-up is closed manually. It only works on web, though, and could be a bit faster. I will look into it, but I would leave it outside of this PR.